### PR TITLE
Feature/connect user data to dom

### PR DESCRIPTION
**Pull request by Wperricone**
#### What's this PR do?
- It worked on trying to fix the bug of being able to add the same recipe to favorites more than once

#### Where should the reviewer start?
-User.js,  scripts.js

#### How should this be manually tested?
- by adding recipes to 'recipes to cook' and 'favorites'

#### Questions:
- how do we fix the bug that allows the user to add the same recipe to favorites more than once?

### CODE REVIEW

File: src/classes/RecipeRepository.js [hunk 1]
```diff
@@ -14,11 +28,15 @@ getRecipesBySearch(input) {
     const recipes = recipesByTag.map((recipe) => {
     recipesByName.map((recipe2) => {
       if(recipe.id === recipe2.id) {
-        recipesByTag.splice([recipe], 2)
-      }
+        return
+        //recipesByTag.splice(recipe, 1)
```
Comment (sted1994) [inline src/classes/RecipeRepository.js:32]: We can delete this

File: src/classes/User.js [hunk 2]
```diff
@@ -5,21 +5,69 @@ export class User {
...
+  })
+}
   };
 
   removeFav(recipe) {
     const index = this.favRecipes.indexOf(recipe);
     this.favRecipes.splice(index, 1)
   };
 
+  searchFavs(input) {
```
Comment (zwsaile) [inline src/classes/User.js:42]: This is where everything goes awry

File: src/classes/User.js [hunk 3]
```diff
@@ -5,21 +5,69 @@ export class User {
...
-    this.recipesToCook.push(recipe)
+    if(!this.recipesToCook.includes(recipe)) {
+      this.recipesToCook.push(recipe)
+    }
   };
 
-  addToFavRecipes(recipe) {
-    this.favRecipes.push(recipe)
+  addToFavRecipes(favRecipe) {
+  //   if(this.favRecipes.includes(recipe)) {
```
Comment (sted1994) [inline src/classes/User.js:18]: we can delete these lines

File: src/classes/User.js [hunk 4]
```diff
@@ -5,21 +5,69 @@ export class User {
...
+      }
+    })
+
+    const recipesByName = this.favRecipes.forEach((recipe) => {
+    let upperCaseInput = input[0].toUpperCase() + input.slice(1)
+          if(recipe.name.includes(input)) {
+        this.filteredFavs.push(recipe)
+      }
+    })
+    // console.log("KINGSLEY", recipesByTag[0]);
```
Comment (sted1994) [inline src/classes/User.js:55]: we can delete these

**Review by sted1994 — APPROVED** [review]

**Review by zwsaile — APPROVED** [review]

**Pull request merged**